### PR TITLE
[Relations/DynamicZones]: manage refetching and cleanData

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

- [x] handles refetching after reordering so if you remove a relation and then reorder the dynamic zone that old relation won't appear
- [x] handles the cleanData so we still disconnect and connect the correct relations even if you've reordered the dynamic zone

### CODE REVIEW

File: examples/getstarted/src/api/kitchensink/content-types/kitchensink/schema.json [hunk 1]
```diff
@@ -156,6 +156,13 @@
       "type": "customField",
       "regex": "^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$",
       "customField": "plugin::color-picker.color"
+    },
```
Comment (jhoward1994) [inline examples/getstarted/src/api/kitchensink/content-types/kitchensink/schema.json:159]: did you mean to commit changes to the example files?

Comment (joshuaellis) [reply examples/getstarted/src/api/kitchensink/content-types/kitchensink/schema.json:159]: I think we should, it's getting tiresome making the same component over and over again every time we look at this area 🥲

File: packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js [hunk 2]
```diff
@@ -155,14 +155,28 @@ const reducer = (state, action) =>
         const initialDataRelations = get(state, initialDataPath);
         const modifiedDataRelations = get(state, modifiedDataPath);
 
-        set(draftState, initialDataPath, uniqBy([...value, ...initialDataRelations], 'id'));
+        /**
+         * Check if the values we're loading are already in initial
+         * data if they are then we don't need to load them at all
+         */
+        const valuesToLoad = value.filter((relation) => {
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js:162]: Nice 💡 that makes sense!

**Comment by joshuaellis** [comment]
@jhoward1994 this is ready for review now.

File: packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/utils/cleanData.js [hunk 3]
```diff
@@ -20,10 +23,12 @@ const cleanData = ({ browserState, serverState }, currentSchema, componentsSchem
    * @param {object} browserState – the modifiedData from REDUX
    * @param {object} serverState – the initialData from REDUX
    * @param {*} schema
+   * @param {string} path
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/utils/cleanData.js:26]: Should we add the correct JSDOC for `pathToParent` ?

Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/utils/cleanData.js:26]: Ah i didn't change the name, yep good suggestion. I'll make the change.

Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/utils/cleanData.js:26]:
```suggestion
   * @param {string} pathToParent - the path to the parent of the current entry
```

**Review by jhoward1994 — APPROVED** [review]
This looks really good, thank you 🙏🏻. I'll sort out the failing snapshot in the other PR

**Pull request merged**